CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,107 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+
+
+# Function for opening the file
+def CAN_Decoder():
+    lineNum = 0
+    oldDataFile = filedialog.askopenfile(initialdir="/")
+    listedOldData = []
+    for dataLine in oldDataFile:
```

Comment:
try using Regex for extracting information from a text file. 
Regex is de facto standard for parsing text and is a must know library in most of programming languages.
This is a good start: https://www.w3schools.com/python/python_regex.asp
you can practice here: https://regex101.com/
If you need help with the complete regex for extracting the information, just let me know

Reply:
Thanks for the resources shared. I have gone through the resources but I am not sure how easy we can use it for our case.